src/wallet/wallet.cpp

```diff
@@ -723,6 +723,23 @@ void CWallet::AddToSpends(const uint256& wtxid)
         AddToSpends(txin.prevout, wtxid);
 }
 
+void CWallet::AddToAddressBloomFilter(const CWalletTx& wtx)
+{
+    for (const auto& output : wtx.tx->vout) {
+        m_address_bloom_filter.insert(ToByteVector(output.scriptPubKey));
+    }
+}
+
+void CWallet::BuildAddressBloomFilter()
+{
+    m_address_bloom_filter_elems = std::max<size_t>(100, mapWallet.size() * 2);
```

Comment:
Is this code assuming that there are two outputs per transaction? Bloom filter performance is very much dependent on this estimate being correct (if you take a filter designed for N elements with fprate=0.000001, and put 2N elements in it, the resulting fprate is only 0.0032).

Reply:
The intent was to not need rebuilding until the wallet doubled in size, but I guess it fails to consider transactions having multiple outputs.

Would another factor (or variable) be better here?

Reply:
Made it 6x for now.

Reply:
I'm still seeing `* 2`. Move the constants `ADDR_BLOOM_FILTER_TX_TO_ELEMENTS_FACTOR` and `ADDR_BLOOM_FILTER_FP_RATE` defined in 39f49f653541ce172a91a8d6996f44831f9a8df2 to  a13c30296f0507b1c6cdd70e42e693fe38d3c9df?